File: backend/src/Customer.kt
package challenge

import java.time.LocalDateTime


class Address

data class Discount(val value: Double)

data class Membership(val date: LocalDateTime, val name: String)

class Customer(val name: String, val email: String) {
    private var discount: Discount? = null

    private val memberships = mutableListOf<Membership>()

    val allMemberships
        get() = memberships

    fun addDiscount(discount: Discount) {
        this.discount = discount
    }

    fun addMembership(product: Product) {
        this.memberships.add(Membership(LocalDateTime.now(), product.name))
    }

}

File: backend/customer.py
from membership import Membership

class Customer:

    def __init__(self, name, email):
        '''
        Constructor for Customer class.

        Args:
            name (string): name of the customer.
            email (string): email address of the customer.
        '''
        self.name = name
        self.email = email
        self.membership = Membership()
        self.billing_address = None
        self.shipping_address = None

    def notify_user(self, order):
        '''
        Notifies user about the order via email.

        Args:
            order (Order): order to notify the user about.
        '''
        pass